### ISSUE TITLE
Plot title should be shifted up when xticks are set to the top of the plot

### ISSUE THREAD

**Issue by ppurka** [matplotlib/matplotlib#1415]
This is an issue that came up in http://trac.sagemath.org/13625

When `subplot.xaxis.tick_top()` is used, then the position of the plot title is not recomputed. Ideally, it should be shifted further up. Example code is as follows

``` python
from matplotlib.figure import Figure
figure = Figure()
subplot = figure.add_subplot(111)
subplot.plot(range(10),range(10))
subplot.xaxis.tick_top()
subplot.set_title('a title')
from matplotlib.backends.backend_agg import FigureCanvasAgg
figure.set_canvas(FigureCanvasAgg(figure))
figure.savefig('/tmp/a.png')
```

**Comment by petehuang** [comment]
I reproduced with 1.5.3:

![temp](https://cloud.githubusercontent.com/assets/1134834/21578460/710717a6-cf35-11e6-812b-dc4acb2210e6.png)

Recommend labeling as geometry manager to mirror #1109, which is still the source of truth for this work

**Comment by efiring** [comment]
I think that if there is going to be a partial fix for this any time soon, it will be via using the same logic for the titles as we are using for the xlabel and ylabel.  They are already taking into account the ticks and tick labels.  An additional wrinkle might be the case of twinned axes; the title-positioning logic for either axes would have to check for ticks and tick labels on both axes.

**Comment by afvincent** [comment]
Closed by #9498 (commit 7b30275).
